<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8" />
	<link rel="stylesheet" type="text/css"  href="../css_style/style.css">
	<title>Home</title>
</head>

<body>
	<header><h1>MetaCLADE</h1></header>
	<div id="content">
	<nav>
		<h2>Menu</h2>
		<ul class = 'menu_nav'>
			<li><a class = "Menu" href="home.html">Home</a></li>
			<li><a class = "Menu" href="small_annotation.html">Few domains annotation</a></li>
			<li><a class = "Menu" href="large_annotation.html">All domains annotation</a></li>
			<li><a class = "Menu" href="references.html">References</a></li>
			<li><a class = "Menu" href="help.html">Help</a></li>
		</ul> 
	</nav>

	<section>
	<h2> Home </h2>
    <h3>This program is a multi-source domain annotation for quantitative <em>metagenomic and metatranscriptomic</em> functional profiling.</h3>
		<p class = 'text'>
		Biochemical and regulatory pathways have until recently been thought and modelled within one cell type, one organism, one species. This vision is being dramatically changed by the advent of whole microbiome sequencing studies, revealing the role of symbiotic microbial populations in fundamental biochemical functions. The new landscape we face requires the reconstruction of biochemical and regulatory pathways at the community level in a given environment. In order to understand how environmental factors affect the genetic material and the dynamics of the expression from one environment to another, one wishes to quantitatively relate genetic information with these factors. For this, we need to be as precise as possible in evaluating the quantity of gene protein sequences or transcripts associated to a given pathway. We wish to estimate the precise abundance of protein domains, but also recognise their weak presence or absence.
		</p>
		<p>
		We introduce MetaCLADE2, and improved profile-based domain annotation pipeline based on the multi-source domain annotation strategy. It provides a domain annotation realised directly from reads, and reaches an improved identification of the catalog of functions in a microbiome. MetaCLADE2 can be applied to either metagenomic or metatranscriptomic datasets as well as proteomes.
		</p>
	</section>
	</div>
	<footer>
		<a href="https://www.sorbonne-universite.fr/" target="_blank"><img id="UPMC-logo" class="img-responsive" src="../server_images/logo_sorbonne.png" alt="" style="max-height: 100px; margin-left: 25px; margin-right: 25px; margin-top: 10px;"></a>

		<a href="http://www.cnrs.fr" target="_blank"><img id="CNRS-logo" class="img-responsive" src="../server_images/cnrs_logo.png" alt="" style="max-height: 100px; margin-left: 25px; margin-right: 25px; margin-top: 10px;"></a>

		<a href="http://www.lcqb.upmc.fr/" target="_blank"><img id="LCQB-logo" class="img-responsive" src="../server_images/lcqb.jpg" alt="" style="max-height: 100px; margin-left: 25px; margin-right: 25px; margin-top: 10px;"></a>
	</footer>
</body>
</html>

